pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java

```diff
@@ -266,10 +267,20 @@ public static String prettyPrint(Expression expression) {
   }
 
   public static String getTableName(PinotQuery pinotQuery) {
-    while (pinotQuery.getDataSource().getSubquery() != null) {
-      pinotQuery = pinotQuery.getDataSource().getSubquery();
+    return getTableName(pinotQuery.getDataSource());
+  }
+
+  public static String getTableName(DataSource dataSource) {
+    if (dataSource.getSubquery() != null) {
+      return getTableName(dataSource.getSubquery());
+    }
+    if (dataSource.getTableName() != null) {
+      return dataSource.getTableName();
+    }
```

Comment:
I am not sure why first and second conditions / cases are being treated as mutually exclusive in a couple of places in the code.